### PR TITLE
Add required_provider blocks

### LINKED ISSUE #42: Missing `aws` provider explicitly specified for this repo
While updating to `tf14`, we found that the upgrade script did not add any `required_providers` block. We then realised it's because there were no `provider` blocks in the original code. 

It might be best to add a specific version provider for `aws` instead of relying on the default one, when not specified.

Context: See [TicketJam notes](https://www.notion.so/gruntwork/TicketJam-Notes-6364138a290444258d91d6b73e15539b#e92d4481e9e045e5b28776938030a246) & [PR open for the TF 14 upgrade script fixes](https://github.com/gruntwork-io/prototypes/pull/115)

### PR DESCRIPTION
**Pull request by infraredgirl**
Closes #42.

### CODE REVIEW

File: modules/executable-dependency/main.tf [hunk 1]
```diff
@@ -3,6 +3,13 @@ terraform {
   # 0.12.26 as the minimum version, as that version added support for required_providers with source URLs, making it
   # forwards compatible with 0.13.x code.
   required_version = ">= 0.12.26"
+
+  required_providers {
+    external = {
+      source  = "hashicorp/external"
+      version = "2.0.0"
```
Comment (brikis98) [inline modules/executable-dependency/main.tf:10]: Do we want to be pinned to an _exact_ version here?

Comment (brikis98) [reply modules/executable-dependency/main.tf:10]: Same question for all the other providers pinned to a specific version, rather than, say, `~> 2.0.0`.

**Review by brikis98 — COMMENTED** [review]
Sanity check for #42: do we _want_ all providers pinned to specific versions in our modules? We certainly want our customers to do so in their services (i.e., root modules), so the only question is whether in our library of reusable modules it's desirable to pin versions?

First pass:

## Advantages

* Assuming we pin within a specific major version number, we don't accidentally pull in backwards incompatible changes.
* Explicit may be better than implicit?

## Drawbacks

* We'll have to go through an upgrade process for every single provider release. 
* Customers won't be able to use new provider versions, even if the upgrade was backwards compatible with how they are being used in our modules, until we've updated our whole library, which can take weeks or months.
* More code for us to maintain.

**Comment by infraredgirl** [comment]
> Sanity check for #42: do we _want_ all providers pinned to specific versions in our modules? We certainly want our customers to do so in their services (i.e., root modules), so the only question is whether in our library of reusable modules it's desirable to pin versions?

Ah, this is a very good point I overlooked - other modules in our library don't have this either, it's the one level higher (services) that have this. IMO the drawbacks here outweigh the advantages, so we probably should **not** add this.

I'll leave this PR open for a bit to give people a chance to disagree, but if no one comes up with a compelling argument why we should do this despite the drawbacks, I'll close this PR.

**Comment by brikis98** [comment]
> > Sanity check for #42: do we _want_ all providers pinned to specific versions in our modules? We certainly want our customers to do so in their services (i.e., root modules), so the only question is whether in our library of reusable modules it's desirable to pin versions?
> 
> Ah, this is a very good point I overlooked - other modules in our library don't have this either, it's the one level higher (services) that have this. IMO the drawbacks here outweigh the advantages, so we probably should **not** add this.
> 
> I'll leave this PR open for a bit to give people a chance to disagree, but if no one comes up with a compelling argument why we should do this despite the drawbacks, I'll close this PR.

Note that pinning the AWS provider to a major version number (e.g., 3.x) _is_ something we do... But for the other providers, it's not clear we need it.

**Comment by infraredgirl** [comment]
Closing this as it's not clear we need this. Happy to reopen if needed!

**Pull request closed without merging**